### PR TITLE
Tweak overflow and inverse thresholds

### LINKED ISSUE #69: Revisit inverse threshold of 1000
Right now, we permit taking inverses of integers as long as the value we divide into is at least 1,000.  This works well for toy examples like the following:

```cpp
inverse_as(hertz, milli(seconds)(5)); // Yields hertz(200)
```

However, it gets pretty bad for "medium" and "large" values.

```cpp
inverse_as(hertz, milli(seconds)(73));  // Yields hertz(13), instead of ~13.699 Hz
inverse_as(hertz, milli(seconds)(251)); // Yields hertz( 3), instead of ~ 3.984 Hz (!)
```

In the overflow safety surface, we used the intuition that we want to be very safe for values up to 1,000, because users would tend to use the next prefixed unit for values much bigger than this.  (We've since raised the threshold to over 4,000 in that case.)

Perhaps a similar intuition here would yield a threshold of 1,000,000 (inclusive)?  In that case, none of the above examples would compile (with integer inputs).  However, `micro(seconds)` would work.  If we had these same underlying values for units of microseconds instead of milliseconds, we would obtain:

```cpp
                                         // Result          True answer
inverse_as(hertz, micro(seconds)(5));    // hertz(200'000)  200,000.0 Hz
inverse_as(hertz, micro(seconds)(73));   // hertz( 13,698)   13,698.6 Hz
inverse_as(hertz, micro(seconds)(251));  // hertz(  3,984)    3,984.1 Hz
```

(Of course, with floating point inputs all of the above would work and would give accurate answers --- this issue is purely about tuning the safety surface for calculations that never leave the integer domain.)

### PR DESCRIPTION
**Pull request by chiphogg**
The primary motivation is to move to a more defensible policy for
inversions.  Formerly, our policy was "avoid truncating nonzero values
to 0" for any value <= 1000.  The new policy is "keep round-trip
inversion lossless" for any such value.  This amounts to changing the
threshold from 1,000 to 1,000,000.

Unfortunately, if we made this change alone, we'd have poor ergonomics
even for pretty reasonable conversions.  For example:

```cpp
constexpr auto period = inverse_as(micro(seconds), hertz(40));
```

This fails because converting `1` to the (dimensionless) product of the
units multiplies by 1,000,000.  Since we're using `int`, this would
overflow the old threshold of 4,294.  So, these thresholds are
connected.

I've actually been thinking for a while that it would be nice to loosen
that threshold ever so slightly.  It's a nuisance to treat `int32_t` and
`uint32_t` differently, despite a very small difference in their ranges.
So I also make that change in this PR.  Ultimately, I think this will be
a very durable "sweet spot".  It's pleasing that we prioritize values of
up to 1,000 generally, and you can fit 3 factors of 1,000 in an `int`,
with a little room to spare (which we put towards extending the safety
surface).

Fixes #69.

Test plan
---------

- [x] Existing unit tests (subject to appropriate tweaks).
- [x] Tested on Aurora's whole codebase.  No significant errors
      introduced.
- [x] Checked compile time impact: no significant effect.
- [x] Added a test, `QuantityPoint.PreservesRep`, which I had wanted to
      add in #101, but couldn't.

### CODE REVIEW

File: au/math_test.cc [hunk 1]
```diff
@@ -713,8 +713,8 @@ TEST(CeilIn, SupportsDifferentOutputTypes) {
 }
 
 TEST(InverseAs, HandlesIntegerRepCorrectly) {
-    constexpr auto period = inverse_as(milli(seconds), hertz(40));
```
Comment (geoffviola) [inline au/math_test.cc:716]: Comment: I guess we are using the integer division operator so there may be data loss for cases like `hertz(41)`. With `hertz(40)`, 1/40 is a multiple of 0.005, but that can only be determined at compile time.

File: au/math_test.cc [hunk 2]
```diff
@@ -713,8 +713,8 @@ TEST(CeilIn, SupportsDifferentOutputTypes) {
...
-    EXPECT_THAT(period, SameTypeAndValue(milli(seconds)(25)));
+    constexpr auto period = inverse_as(micro(seconds), hertz(40));
```
Comment (geoffviola) [inline au/math_test.cc:716]: Comment: This PR mentions round trip data loss, but operations like `inverse_as(hertz, micro(seconds(25'000)))` don't compile.

**Review by geoffviola — APPROVED** [review]
Round-trip lossless inversion sounds like a good direction. The tweaks to align `int32_t` to work better with micro seems reasonable. Thanks!

File: au/math_test.cc (hunk 2 above)
Comment (chiphogg) [reply au/math_test.cc:716]: Misplaced `)`.  This compiles:

```cpp
inverse_as(hertz, micro(seconds)(25'000))
```

It hadn't occurred to me before that people might try applying a prefix to a quantity, not a unit, but I can see how this syntax error would be an easy one to make.  :thinking:

File: au/math_test.cc (hunk 1 above)
Comment (chiphogg) [reply au/math_test.cc:716]: `hertz(41)` is technically still lossless in the sense that the round trip is guaranteed to recover the initial value for everything up to 1,000.  (It's true that the integer result itself won't be exact, of course.)

**Pull request merged**